runtimes/adapters/e2b.py

```
@@ -0,0 +1,365 @@
+        """Run ``pytest`` inside the sandbox. Returns ``(output, passed)``.
+
+        Best-effort: if pytest is not installed in the sandbox template, or
+        there are no tests, returns ``("", True)`` so the verifier is a no-op
+        rather than a hard failure.
+        """
+        try:
+            raw = await session.call_tool("run_command", {
+                "cmd": "cd repo && python -m pytest -x --tb=short 2>&1 | tail -200",
+                "timeout": 120,
+            })
+        except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)